lib/util.coffee

```diff
@@ -23,3 +27,73 @@ exports.treatAsMissingDevice = (uuidOrId) ->
 		replacementErr = new errors.ResinDeviceNotFound(uuidOrId)
 		replacementErr.stack = err.stack
 		throw replacementErr
+
+# Merging two sets of pine options sensibly is more complicated than it sounds.
+#
+# The general rules are:
+# * select, orderby, top and skip override (select this, instead of the default)
+# * filters are combined (i.e. both filters must match)
+# * expands are combined (include both expansions), and this recurses down.
+#   * That means $expands within expands are combined
+#   * And $selects within expands override
+# * Any unknown options 'extra' options throw an error. Unknown 'default' options are ignored.
+exports.mergePineOptions = (defaults, extras) ->
+	if not extras
+		return defaults
+
+	result = cloneDeep(defaults)
+
+	for own option, value of extras
+		switch option
+			when 'select', 'orderby', 'top', 'skip'
+				result[option] = value
+
+			when 'filter'
+				if defaults.filter
+					result.filter = $and: [defaults.filter, value]
+				else
+					result.filter = value
+
+			when 'expand'
+				result.expand = mergeExpandOptions(defaults.expand, value)
+
+			else
+				throw new Error("Unknown pine option: #{option}")
```

Comment:
> at the point we update the minimum dep version and start using them then we would add any special case handling needed here as well

But when we do this, if we do need any special case handling, it's always a breaking change. As we start using the sdk more and more (e.g. in the ui, and in etcher), breaking changes will be a reasonable size annoyance in quite a few places (not to mention for sdk end users).

Feels like that's unnecessary - we're allowing things in the api that may or may not come back to cause problems for us, to save the tiny bit of effort to add options to the whitelist here if we want them (if we do ever add any at all). I'd rather be cautious. If we don't have to worry about breaking changes, we can add new options here with impunity and all downstream clients will immediately get those for free, so everybody gets less friction all round.

Also has the benefit of catching you if you try to use options that don't exist, e.g. if you pass in something with a typo or similar.

I think in practice adding more pine options here looks unlikely to come up very often, and I don't hate this, so if you do feel strongly about it we can go down this road, but I'd definitely prefer the cautious option by default.